lib/ohai/plugins/solaris2/platform.rb

```diff
@@ -38,17 +38,15 @@
     file_open("/etc/release") do |file|
       while ( line = file.gets )
         case line
-        when /^.*(SmartOS).*$/
+        when /.*SmartOS.*/
           platform "smartos"
-        when /^\s*(OmniOS).*r(\d+).*$/
+        when /^\s*OmniOS.*r(\d+).*$/
           platform "omnios"
-          platform_version $2
-        when /^\s*(OpenIndiana).*oi_(\d+).*$/
+          platform_version $1
+        when /^\s*OpenIndiana.*(Development oi_|Hipster )(\d\S*)/ # https://rubular.com/r/iMtOBwbnyqDz7u
           platform "openindiana"
           platform_version $2
-        when /^\s*(Oracle Solaris)/
-          platform "solaris2"
-        when /^\s*(Solaris)\s.*$/
+        when /^\s*(Oracle Solaris|Solaris)/
```

Comment:
Why not just `/Solaris/` at this point? Oracle Solaris. Sun Solaris. Solaris... anything that's not the above and has `Solaris` in it, probably should match Solaris.

That said, I see no specific issue with what you have, just seems like more generic catch all would be better.

But if you really prefer to be specific that's cool. Or you could also shorten it to `/^\s*(Oracle )?Solaris/`

Reply:
It made me nervous to open it up to that, but I guess as the last match we were doing that anyways

Reply:
Thinking about it more I still think we want to match on whitespace and then Oracle Solaris or Solaris so this is probably the right thing to do here.